test/python/quantum_info/operators/symplectic/test_sparse_pauli_op.py

```diff
@@ -819,6 +819,7 @@ def test_simplify_zero(self, num_qubits):
         np.testing.assert_array_equal(zero_op.paulis.phase, np.zeros(zero_op.size))
         np.testing.assert_array_equal(simplified_op.paulis.phase, np.zeros(simplified_op.size))
 
+    @unittest.skipUnless(optionals.HAS_SYMPY, "sympy required")
```

Comment:
Why do these tests need sympy?

Reply:
Internally `.simplify()` they calls`. sympify()`: https://github.com/Qiskit/qiskit/blob/343cf09c4167799554d2218feb04fd5f21de137d/qiskit/quantum_info/operators/symplectic/sparse_pauli_op.py#L485 and then runs Sympy's `is_Number`. We probably can refactor it to do the check for bound parameters directly without sympy, but I figured that probably should be done as a standalone pr.

Reply:
Huh, but `SparsePauliOp.simplify` seems like a pretty standard thing to call that we still need sympy for. Do you think it can still be optional if it's needed here?